File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/10 - 2 and 3 May 2019/03. Mobile operator.py
contract_years = input()
contract_type = input()
internet = input()
months = int(input())

month_price = 0

if contract_years == "one":
    if contract_type == "Small":
        month_price = 9.98
    elif contract_type == "Middle":
        month_price = 18.99
    elif contract_type == "Large":
        month_price = 25.98
    elif contract_type == "ExtraLarge":
        month_price = 35.99
elif contract_years == "two":
    if contract_type == "Small":
        month_price = 8.58
    elif contract_type == "Middle":
        month_price = 17.09
    elif contract_type == "Large":
        month_price = 23.59
    elif contract_type == "ExtraLarge":
        month_price = 31.79

if internet == "yes":
    if month_price <= 10:
        month_price += 5.5
    elif month_price <= 30:
        month_price += 4.35
    elif month_price > 30:
        month_price += 3.85

total = month_price * months

if contract_years == "two":
    total -= total * 0.0375

print(f"{total:.2f} lv.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/02.  Online Retake Exam - 2 and 3 May 2019/03. Mobile operator.js
function mobileOperator(params) {
    let period = params.shift();
    let contract = params.shift();
    let internet = params.shift();
    let months = Number(params.shift());

    let prices = {
        'one': {'Small': 9.98, 'Middle': 18.99, 'Large': 25.98, 'ExtraLarge': 35.99},
        'two': {'Small': 8.58, 'Middle': 17.09, 'Large': 23.59, 'ExtraLarge': 31.79}
    };
    
    let price = prices[period][contract];
    if (internet == 'yes') {
        if (price <= 10) {
            price += 5.5;
        } else if (price > 10 && price <= 30) {
            price += 4.35;
        } else if (price > 30) {
            price += 3.85;
        }
    }
    if (period == "two") {
        price *= 0.9625;
    }
    price *= months;

    console.log(`${price.toFixed(2)} lv.`);
}

mobileOperator(['one', 'Small', 'yes', '10']);  // Expected Output: 154.80 lv.
mobileOperator(['two', 'Large', 'no', '10']);  // Expected Output: 227.05 lv.
mobileOperator(['two', 'ExtraLarge', 'yes', '20']);  // Expected Output: 686.07 lv.
mobileOperator(['two', 'Small', 'yes', '20']);  // Expected Output: 271.04 lv.
